app/src/main/java/me/guillaumin/android/osmtracker/db/TracklistAdapter.java

```diff
@@ -20,9 +24,52 @@
  *
  */
 public class TracklistAdapter extends CursorAdapter {
+	
+	private static final float STOP_SHOWING_DECIMALS_AFTER = 20;
+	
+	/** The TrackManager who created us */
+	private TrackManager owner;
+
+	public static String distanceToString(float distance, Resources resources) {
+		if (distance < 100) {
+			// Exact meters
+			return resources.getString(R.string.trackmgr_distance_in_meters).replace("{0}",
+					String.valueOf(Math.round(distance)));
+		} else if (distance < 1000) {
+			// Round to 10 meters
+			return resources.getString(R.string.trackmgr_distance_in_meters).replace("{0}",
+					String.valueOf(10 * Math.round(distance / 10)));
+		} else if (distance < STOP_SHOWING_DECIMALS_AFTER*1000) {
+			// Kilometers with 1 decimal
+			return resources.getString(R.string.trackmgr_distance_in_kilometers).replace("{0}",
+					String.format("%.1f", distance / 1000));
+		} else {
+			// Whole kilometers
+			return resources.getString(R.string.trackmgr_distance_in_kilometers).replace("{0}",
+					String.valueOf(Math.round(distance / 1000)));
+		}
+	}
+
+	public static String speedToString(float speed, Resources resources) {
+		float kmph = (float)3.6*speed; // convert meters per second to kilometers per hour
+		if (kmph < 1) {
+			// Round to one non-zero digit
+			return resources.getString(R.string.trackmgr_speed_in_kmph).replace("{0}",
+					String.format("%.1g", kmph));
+		} else if (kmph < STOP_SHOWING_DECIMALS_AFTER) {
+			// Round to one decimal
+			return resources.getString(R.string.trackmgr_speed_in_kmph).replace("{0}",
+					String.format("%.1f", kmph));
+		} else {
+			// Round to integer
+			return resources.getString(R.string.trackmgr_speed_in_kmph).replace("{0}",
+					String.valueOf(Math.round(kmph)));
+		}
+	}
 
 	public TracklistAdapter(Context context, Cursor c) {
 		super(context, c);
+		owner = (TrackManager) context;

```

Comment:
That adds a dependency against the container of the view, meaning that we can't use `TracklistAdapter` from another activity. It would be nice to avoid